nixos/modules/services/misc/weechat.nix

```diff
@@ -0,0 +1,103 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.weechat;
+in
+
+{
+  options.services.weechat = {
+    enable = mkEnableOption "weechat";
+    init = mkOption {
+      description = "Weechat commands applied at start, one command per line.";
+      example = ''
+        /set relay.network.password correct-horse-battery-staple
+        /relay add weechat 9001
+      '';
+      type = types.str;
+      default = "";
+    };
+    root = mkOption {
+      description = "Weechat state directory.";
+      type = types.str;
+      default = "/var/lib/weechat";
+    };
+    scripts = mkOption {
+      description = "Custom scripts to be loaded by the weechat module.";
+      type = with types; listOf (submodule {
+        options = {
+          name = mkOption {
+            type = str;
+            description = "Path to the script in the derivation.";
+          };
+          script = mkOption {
+            type = types.package;
+            description = "Package which contains the actual script.";
+          };
+        };
+      });
+      default = [];
+      example = literalExample ''
+        [
+          { name = "share/jabber.py"; script = pkgs.weechat-xmpp }
+        ]
+      '';
+    };
+    sessionName = mkOption {
+      description = "Name of the `screen' session for weechat.";
+      default = "weechat-screen";
+      type = types.str;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users = {
+      groups.weechat = {};
+      users.weechat = {
+        createHome = true;
+        group = "weechat";
+        home = cfg.root;
+        isSystemUser = true;
+      };
+    };
+
+    systemd.services.weechat = {
+      environment.WEECHAT_HOME = cfg.root;
+      serviceConfig = {
+        User = "weechat";
+        Group = "weechat";
+        RemainAfterExit = "yes";
+      };
+      script = "exec ${pkgs.screen}/bin/screen -Dm -S ${cfg.sessionName} ${pkgs.weechat}/bin/weechat";
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network.target" ];
+    };
+
+    systemd.paths.weechat-fifo = {
+      pathConfig = {
+        PathExists = "${cfg.root}/weechat_fifo";
+        Unit = "weechat-apply-init.service";
+      };
+      wantedBy = [ "multi-user.target" ];
+    };
+
+    services.weechat.init = mkIf (builtins.length cfg.scripts != 0) (concatStringsSep "\n"
+      (map (script: "/script load ${script.script}/${script.name}\n") cfg.scripts));
+
+    systemd.services.weechat-apply-init = let
+      initFile = pkgs.writeText "weechat-init" cfg.init;
+    in {
+      script = "sed 's/^/*/' ${initFile} > ${cfg.root}/weechat_fifo";
```

Comment:
What does this do?

Reply:
according to the docs (https://weechat.org/files/doc/stable/weechat_user.en.html#fifo_plugin), the `*` at the beginning of a string is needed when passing commands to weechat through `weechat_fifo`. This was actually introduced in the previous commit I didn't author, but it's more convenient to type `/cmd args...` without the star at the beginning.

Reply:
I just used the code from the previous commit. However I think that this is the right way to go as the star is only used with `fifo`.
However the `/` at the beginning is also used when running weechat from an active session, so it's IMHO more convenient to keep this.